src/mount-support.c

```diff
@@ -293,3 +294,55 @@ void setup_slave_mount_namespace()
 		die("can not make make / rslave");
 	}
 }
+
+void setup_bind_mounts(const char *appname)
+{
+	debug("%s: %s", __FUNCTION__, appname);
+
+	FILE *f = NULL;
+	const char *bind_profile_dir = "/var/lib/snapd/bind/profiles/";
+
+	char profile_path[PATH_MAX];
+	int snprintf_rc =
+	    snprintf(profile_path, sizeof(profile_path), "%s/%s.bind",
+		     bind_profile_dir, appname);
+	if (snprintf_rc < 0 || snprintf_rc >= 512) {
+		errno = 0;
+		die("snprintf returned unexpected value");
+	}
+
+	f = fopen(profile_path, "r");
+	// it is ok for the file to not exist
+	if (f == NULL && errno == ENOENT)
+		return;
+	// however any other error is a real error
+	if (f == NULL) {
+		die("cannot open %s", profile_path);
+	}
+
+	struct mntent *m = NULL;
+	while ((m = getmntent(f)) != NULL) {
+		int flags = MS_BIND;
+
+		if (strcmp(m->mnt_type, "") != 0) {
+			die("only bind mounts are supported");
+		}
+		if (hasmntopt(m, "bind") == NULL) {
+			die("need bind mount flag");
+		}
+		if (hasmntopt(m, "ro") != NULL) {
+			flags |= MS_RDONLY;
+		}
+
```

Comment:
You're correct that AppArmor will be confining everything that is bind mounted in so any binaries made available via the bind mount will still be confined. My suggestion was simply based on the idea to have the most secure defaults and then have the interface author have to specify "exec" if they expected binaries to be executed from the bind mount.

I'm ok either way but defaulting to nosuid,nodev while also defaulting to exec may be a bit confusing to interface authors.